src/components/AdminPage/UserForm/AddingUserForm.js

```diff
@@ -0,0 +1,157 @@
+  const [phoneTouched, setPhoneTouched] = useState(false);
+  const isValid = enteredPassword != null && enteredPassword.trim().length > 5;
+
+  const enteredNameHandler = (event) => {
+    setEnteredName(event.target.value);
+  };
+  const enteredEmailHandler = (event) => {
+    setEnteredEmail(event.target.value);
+  };
+  const enteredPhoneHandler = (event) => {
+    const onlyDigits = event.target.value.replace(/\D/g, "");
+    setEnteredPhone(onlyDigits);
+    setPhoneTouched(true);
+  };
+
+  const enteredPasswordHandler = (event) => {
+    setEnteredPassword(event.target.value);
+    setPasswordTouched(true);
+  };
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+
+    const newUserData = {
+      name: enteredName,
+      email: enteredEmail,
+      phone: enteredPhone,
+      password: enteredPassword,
+    };
+
+    addUser(newUserData);
+
+    setEnteredName("");
+    setEnteredEmail("");
+    setEnteredPhone("");
+    setEnteredPassword("");
+    setPasswordTouched(false);
+    setPhoneTouched(false);
+  };
+
+  const isPasswordValid = (event) => {
+    setEnteredPassword(event.target.value);
+  };
+
+  const checkIfPasswordIsValid = () => {
+    if (!passwordTouched) {
+      return true;
+    }
+    return enteredPassword.length > 5;
+  };
+
+  const checkIfPhoneIsTooShort = () => {
+    return phoneTouched && enteredPhone.length < 9;
+  };
+
+  return (
+    <Container style={{ padding: "0" }}>
+      <Form
+        className={` ${styles.inputsContainer} w-100 d-flex flex-column align-items-center p-3`}
+        style={{
+          backgroundColor: "#C4C4C4",
+        }}
+        onSubmit={submitHandler}
+      >
+        <FormGroup style={{ minWidth: "50%" }}>
+          <FormControl
+            className={`p-2 shadow-none text-white ${styles.input}`}
+            style={{ backgroundColor: "#5E5B5B", borderColor: "transparent" }}
+            placeholder="Name and Surname"
+            type="text"
+            value={enteredName}
+            onChange={enteredNameHandler}
+            required
+          ></FormControl>
+          <FormControl
+            className={`mt-2 p-2 shadow-none text-white ${styles.input}`}
+            style={{ backgroundColor: "#5E5B5B", borderColor: "transparent" }}
+            placeholder="E-mail"
+            type="email"
+            value={enteredEmail}
+            onChange={enteredEmailHandler}
+            required
+          ></FormControl>
+          <FormControl
+            className={`mt-2 p-2 shadow-none text-white ${styles.input}`}
+            style={{ backgroundColor: "#5E5B5B", borderColor: "transparent" }}
+            placeholder="Phone number"
+            type="tel"
+            value={enteredPhone}
+            onChange={enteredPhoneHandler}
+            pattern="^[0-9]*$"
+            maxLength={13}
+            required
+          ></FormControl>
+          {checkIfPhoneIsTooShort() && <p>Your phone number is too short</p>}
+          <FormControl
```

Comment:
I think a message would fit here, something like: Incorrect phone number format. Second, why 13 digits? Third, check that the entered data complies with the format. And display the appropriate message here. that is, if it is too short, report it as now, but also add a message about an incorrect format.

Reply:
Sometimes format of phone numbers looks sth like this: 0048124466780  (13 digits), so I established it as the max length.